### ISSUE TITLE
Crash when exiting app

### ISSUE THREAD

**Issue by CactusTheGuy** [mihonapp/mihon#287]
### Steps to reproduce

1. Enter the app
2. Read for like an hour (possibly suspend the app a few times)
3. Exit the app

### Expected behavior

No crash when exiting app

### Actual behavior

Crash when exiting app

### Crash logs

[mihon_crash_logs.txt](https://github.com/mihonapp/mihon/files/14093343/mihon_crash_logs.txt)

### Mihon version

0.16.3

### Android version

MIUI 14 Android 13

### Device

Redmi Note 10 Pro

### Other details

This happen only once a week during usage of the app.
Feel free to bonk me if this annoying...

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I have gone through the [FAQ](https://mihon.app/docs/faq/general) and [troubleshooting guide](https://mihon.app/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.16.3](https://github.com/mihonapp/mihon/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by CactusTheGuy** [comment]
Closing this myself due to it's a duplicate of #282
